Layout.py

```diff
@@ -47,27 +50,53 @@ def fixed_set_layout(self, layout):
     def fixed_focus_group(self, group):
         # I have no idea why this is work instead of
         # `self.window.focus_group(group)`
-        sublime.set_timeout(lambda : self.window.focus_group(group), 0)
-        sublime.set_timeout(lambda : self.window.focus_group(group), 0)
+        sublime.set_timeout(lambda: self.window.focus_group(group), 0)
+        sublime.set_timeout(lambda: self.window.focus_group(group), 0)
+
+    def refresh_layout(self):
+        groups = sublime.active_window().num_groups()
+        current_gr = sublime.active_window().active_group()
+        current_view = sublime.active_window().active_view()
+        for i in range(0, groups):
+            sublime.active_window().focus_group(i)
+        sublime.active_window().focus_group(current_gr)
+        sublime.active_window().focus_view(current_view)
 
     def layout_to_json(self):
         layout = self.layout
         layout['views_in_group'] = []
         layout['active_group'] = self.current_group
 
-        def view_to_jsonable(view):
+        def view_to_jsonable(view, active=False):
             return dict(
                 file_name=view.file_name(),
                 name=view.name(),
                 is_read_only=view.is_read_only(),
-                is_scratch=view.is_scratch()
+                is_scratch=view.is_scratch(),
+                is_active=active
             )
 
+        def is_active(active, current):
+            if active == current:
+                return True
+            else:
+                return False
+
         for i in range(len(layout['cells'])):
             views = self.window.views_in_group(i)
-            views = [view_to_jsonable(view) for view in views]
+            active_view = self.window.active_view_in_group(i)
+            views = [view_to_jsonable(view, is_active(
+                active_view, view)) for view in views]
             layout['views_in_group'].append(views)
 
+        for group in range(len(layout['cells'])):
+            views = layout['views_in_group'][group]
+            for view in views:
+                if view['is_active']:
+                    active = view
+                    views.remove(view)
+            views.append(active)
```

Comment:
removing the view and adding it again after loop is complete is intentional, if I simply give focus while the loop is running, the new views will take the focus, instead I pull out the active view and I append it again at the end of the list for that group, so that when the layout file is opened the active view is opened last and receives the focus